design/one-pager-stacks-security-isolation.md

```diff
@@ -241,13 +281,258 @@ permissionScope: Cluster
+ scope that will facilitate secure end user interaction with Crossplane resources. Every stack install provides CRDs
+ which require updating the roles in the scope of any given install to be able to consume those resources. Instead of
+ end-users needing to manually update ClusterRoles, or Roles, the stack-manager will
+ [update the permissions](#permission-aggregation-for-roles) of built in roles dynamically as stacks are installed,
+ updated and deleted across cluster or namespace scopes. We are modeling our permissions based on default roles and
+ bindings that kubernetes provides in
+ [built-in kubernetes roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings).
+
+We will model our roles to define a clear separation of concerns for cluster(environment) scoped resources,
+ and namespaced resources. In our model we assume that any Cluster Stack resources are global will be for the
+ environment or infrastructure scope, whereas any namespace stack resources will be for the namespace that that stack
+ is installed into.
+
+At install time:
+ - Cluster Stack: generate a set of `ClusterRole` objects that will be aggregated to the environment
+  admin, edit, view `ClusterRole` objects.
+ - Namespaced Stack: generate a set of `ClusterRole` objects that will be aggregated to the namespace
+  specific admin, edit, view `ClusterRole` objects.
+
+#### Default Crossplane ClusterRoles
+
+The following built in roles will be installed and packaged with Crossplane, and updated with role aggregation as we
+ install additional stacks.
+
+The verbs supported in these roles(admin,edit,view) will model behavior based on existing logic around
+ [built-in kubernetes roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings).
+
+|ClusterRole Name | Permissions |
+|:---------------------|:----------|
+| crossplane-admin | Admin: ClusterStacks, Crossplane CRDs, Namespaces, (Cluster)RoleBindings, (Cluster)Roles |
+| crossplane-env-admin | Admin: ClusterStacks, globally scoped Crossplane CRDs |
+| crossplane-env-edit | Edit: ClusterStacks, globally scoped Crossplane CRDs |
+| crossplane-env-view | View: Globally scoped Crossplane CRDs, Not permitted: secrets |
+
+We will provide a convenience group - crossplane:master - that binds to crossplane-admin with a
+ ClusterRoleBinding. This mirrors Kubernetes functionality for cluster-admin and system:master. At install time you will
+ then be able to impersonate a group that is bound to crossplane-admin.
+
+Example running a command as crossplane-admin
+```
+kubectl {any crossplane-admin operation} --as-group=crossplane:master
+```
+
+For discoverability of the roles for the environment we will add the following labels:
+ * `crossplane.io/scope: "environment"`
+
+Match labels on these roles for auto-aggregation:
+ * `rbac.crossplane.io/aggregate-to-crossplane-{install-scope}-{role}: "true"`
```

Comment:
since this is talking about a cluster/environment role, this match role statement example could be filled in with the `install-scope` right here in this doc, right? it's not variable, is it?

should it just always be "environment", i.e. the value of crossplane.io/scope: "environment" just above?